app/models/template.rb

```diff
@@ -278,4 +236,12 @@ def set_defaults
       self.archived ||= false
       self.links ||= { funder: [], sample_plan: [] }
     end
+    
+    # Only one version of a template should be published at a time, so if this one was published make sure other versions are not
+    def reconcile_published
+      if self.published?
```

Comment:
if the method is intended to be called when the callback is fired and passes the condition, why adding again the if here?